src/subscriptions/SubscriptionManager.cpp

```diff
@@ -446,14 +460,39 @@ SubscriptionManager::unsubProposedAccount(
 void
 SubscriptionManager::subProposedTransactions(std::shared_ptr<WsBase> session)
 {
-    txProposedSubscribers_.subscribe(session);
+    subscribeHelper(
+        session, txProposedSubscribers_, [this](session_ptr session) {
+            unsubProposedTransactions(session);
+        });
 }
 
 void
 SubscriptionManager::unsubProposedTransactions(std::shared_ptr<WsBase> session)
 {
     txProposedSubscribers_.unsubscribe(session);
 }
+void
+SubscriptionManager::subscribeHelper(
+    std::shared_ptr<WsBase>& session,
+    Subscription& subs,
+    CleanupFunction&& func)
+{
+    subs.subscribe(session);
+    std::unique_lock lk(cleanupMtx_);
+    cleanupFuncs_[session].emplace_back(func);
```

Comment:
Here 'func' is an lvalue within the body of the function (even tho it's passed as &&, it has a name so it's an lvalue) so you are actually copying 'func'. Also `emplace_back` does not actually _emplace_ here so i suggest you do `push_back(std::move(func))` instead.